account_invoice_triple_discount/models/account_move_line.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import functools
+
+from odoo import api, fields, models
+from odoo.tools import float_compare
+
+
+class AccountMoveLine(models.Model):
+
+    _inherit = "account.move.line"
+
+    discount2 = fields.Float(string="Discount 2 (%)", digits="Discount",)
+    discount3 = fields.Float(string="Discount 3 (%)", digits="Discount",)
+
+    def create(self, values_list):
+        """
+        During the create of move lines, if the system detect that there is a
+        difference between the balance and the price subtotal, it will update
+        the unit price. When computing those, Odoo base module use a single
+        discount on creation. So as there is a difference of the price given
+        by the UI and the price computed during create method, the system will
+        change the unit price of the invoice line. To avoid that, we update
+        the discount field to have the aggregated discount, and we change it
+        back after the creation.
+        (Similar to _recompute_tax_lines on account.move)
+        """
+        old_values = []
+        dp_discount = self.env["decimal.precision"].precision_get("Discount")
+        for values in values_list:
+            old_discount = values.get("discount")
```

Comment:
```suggestion
            old_discount = values.get("discount", 0.0)
```

Reply:
This will fix the error on installation, with steps:
1. Create new database with demo data.
2. Install module account_invoice_triple_discount without first install account module.

Reply:
Thanks, should be fixed now